File: GroupAnagrams.java
class GroupAnagrams {
    public List<List<String>> anagrams(String[] strs) {
        
        if(strs.length == 0) {
            return new ArrayList();
        }
        
        //create a map where key is the sorted string and value is the list of string
        Map<String, List> map = new HashMap<String, List>();
        
        for(String str : strs) {
            //convert the str to character of array
            char[] ch = str.toCharArray();
            //sort the array
            Arrays.sort(ch);

            //convert the character array to a string and assign the sorted character string as a key
            String key = String.valueOf(ch);
            
            //if key is not present, add the key n value into the map
            if(!map.containsKey(key)) {
                map.put(key, new ArrayList());
            }
            
            map.get(key).add(str);
        }
        return new ArrayList(map.values());
    }
}

File: group_anagrams.py
from collections import defaultdict

class GroupAnagrams:
    def anagrams(self, strs):
        if len(strs) == 0:
            return []
        
        # create a map where the key is the sorted string and the value is the list of strings
        anagram_map = defaultdict(list)
        
        for s in strs:
            # sort the string to use as a key
            key = ''.join(sorted(s))
            
            # add the original string to the list of its corresponding sorted key
            anagram_map[key].append(s)
        
        # return the grouped anagrams as a list of lists
        return list(anagram_map.values())

# Example usage
if __name__ == "__main__":
    strs = ["eat", "tea", "tan", "ate", "nat", "bat"]
    group_anagrams = GroupAnagrams()
    result = group_anagrams.anagrams(strs)
    print(result)
